server.go

```diff
@@ -285,6 +287,31 @@ func (s *Server) load() error {
 	})
 }
 
+func (s *Server) StartSelfMonitoring(database, retention string, interval time.Duration) error {
+	if interval == 0 {
+		return fmt.Errorf("statistics check interval must be non-zero")
+	}
+
+	go func() {
+		for {
+			time.Sleep(interval)
```

Comment:
Since the self-monitoring is a form of instrumentation, we should probably be using a [time.Ticker](http://golang.org/pkg/time/#NewTicker).

For a good demonstration of the difference between the two, compare the output of [using time.Sleep](http://play.golang.org/p/hrXDIaTQBC) vs. [using time.NewTicker](http://play.golang.org/p/-fGPGeVfH1).